utils/blackbox.py

```diff
@@ -0,0 +1,363 @@
+from typing import Callable
+
+from algosdk.v2client import algod
+
+from graviton import blackbox
+
+from pyteal import (
+    Arg,
+    Btoi,
+    Bytes,
+    Expr,
+    Int,
+    Itob,
+    Len,
+    Log,
+    Mode,
+    ScratchVar,
+    Seq,
+    SubroutineFnWrapper,
+    TealType,
+    Txn,
+)
+
+# ---- Clients ---- #
+
+
+def algod_client(
```

Comment:
@tzaffi Minor:  What use cases do you have in mind where it'd be preferable to create a client _without_ asserting it's connected?

* I'm trying to gauge if there's a need to have users disambiguate `algod_client` from `algod_client_with_assertion`?
  * On the surface, I'd imagining using `with_assertion` all the time.
* _If_ we retain both methods, then I think `algod_with_assertion` ought to take the same parameters + defaults as `algod_client`.

Reply:
👍 

`algod_client()` doesn't call out so when accessing an `algod` in a context that doesn't already have it you can save a round trip. At some point in the evolution of this PR, this was actually useful.  However, there isn't any such usage now so I'm going to renaming it to `_algod_client()` indicating that this is private and only used internally (in fact, only by `algod_with_assertion()`. We can always make `_algod_client()` public if this need arises again.

Reply:
@tzaffi Gotcha - Thanks for the change in https://github.com/algorand/pyteal/pull/249/commits/01f681b0ab23abbeb63ac0bb75fb2b87120236a6.   Resolving.